lib/matplotlib/tests/test_axes.py

```diff
@@ -2392,12 +2392,17 @@ def test_scatter_color(self):
                                     {'vmax': 0}
                                 ])
     def test_scatter_color_warning(self, kwargs):
+        warn_match = "'c' is empty or not given, "
         # Warn for cases where 'cmap', 'norm', 'vmin', 'vmax'
         # kwargs are being overridden
-        with pytest.warns():
+        with pytest.warns(Warning, match=warn_match):
             plt.scatter([], [], **kwargs)
+            plt.scatter([1,2], [3,4], c=[], **kwargs)
```

Comment:
It is better to put these each in their own context manager other wise if 1 warns and 1 does not this test still passes!